test/unit/authentication/strategy/token_test.rb

```diff
@@ -1,11 +1,10 @@
 require 'test_helper'
 
-class Authentication::Strategy::SsoTest < ActiveSupport::TestCase
+class Authentication::Strategy::TokenTest < ActiveSupport::TestCase
 
   def setup
     @provider = FactoryBot.create(:provider_account)
-    @account  = FactoryBot.create(:buyer_account, provider_account: @provider)
-    @strategy = Authentication::Strategy.build(@provider)
+    @strategy = Authentication::Strategy.build_strategy(:token).new(@provider, true)
```

Comment:
My understanding is that there is no impersonation of buyers so we should test with a provider. Did I get it right?

Reply:
Yeah exactly, the test suite made no sense before, it was trying the impersonation auth on a buyer, and it was loading the OAuth strategy and relying in the hierarchy traversal to reach the impersonation auth. I made it more clear.